src/server/frontend_xwayland/xwayland_surface.cpp

```diff
@@ -182,10 +184,6 @@ auto wm_window_type_to_mir_window_type(
             // TODO: perhaps desktop and dock should set the layer as well somehow?
             return mir_window_type_freestyle;
         }
-        if (wm_type == connection->_NET_WM_WINDOW_TYPE_DIALOG)
-        {
-            return mir_window_type_dialog;
-        }
```

Comment:
It seems unintuitive that `_NET_WM_WINDOW_TYPE_DIALOG` isn't a dialog and using "gloss" instead doesn't seem justified by a first reading of the spec. (It does work better though!)